src/main/java/eu/mihosoft/vrl/v3d/TextExtrude.java

```diff
@@ -15,136 +44,117 @@
-				points.add(new Vector3d(tmp, tmp1, 0));
-				break;
-			case PathIterator.SEG_LINETO:
-				// println "SEG_LINETO "+coords
-				tmp = coords[0];
-				tmp1 = coords[1];
-				points.add(new Vector3d(tmp, tmp1, 0));
-				break;
-			case PathIterator.SEG_MOVETO:
-
-				// move without drawing
-				start[0] = tmp = coords[0];
-				start[1] = tmp1 = coords[1];
-				// println "Moving to "+start
-				points.add(new Vector3d(tmp, tmp1, 0));
-				break;
-			case PathIterator.SEG_CUBICTO:
-				for (float t = 0.0f; t <= 1.05f; t += 0.1f) {
-					// p = a0 + a1*t + a2 * tt + a3*ttt;
-					float tt = t * t;
-					float ttt = tt * t;
-					float p1 = tmp + (coords[0] * t) + (coords[2] * tt) + (coords[4] * ttt);
-					float p2 = tmp1 + (coords[1] * t) + (coords[3] * tt) + (coords[5] * ttt);
-					points.add(new Vector3d(p1, p2, 0));
-					// println "SEG_CUBICTO "+p1+" and "+p2
-				}
-				tmp = coords[4];
-				tmp1 = coords[5];
-				break;
+		public void setLetter(String letter) {
+			this.letter = letter;
+		}
 
-			default:
-				throw new RuntimeException("Unknown iterator type: " + type);
-			}
-			pi.next();
-			// println "pi.isDone() "+pi.isDone()
+		public boolean isHole() {
+			return hole;
+		}
+
+		public void setHole(boolean isHole) {
+			this.hole = isHole;
+		}
+
+		public List<Vector3d> getPoints() {
+			return points;
+		}
+
+		public void setPoints(List<Vector3d> points) {
+			this.points = points;
+		}
+
+		public Path getPath() {
+			return path;
+		}
+
+		public void setPath(Path path) {
+			this.path = path;
+		}
+
+		public Vector3d getOrigen() {
+			return origen;
+		}
+
+		public void setOrigen(Vector3d origen) {
+			this.origen = origen;
+		}
+
+		public List<LineSegment> getHoles() {
+			return holes;
+		}
+
+		public void setHoles(List<LineSegment> holes) {
+			this.holes = holes;
 		}
+
+		public void addHole(LineSegment hole) {
+			holes.add(hole);
+		}
+
+		@Override
+		public String toString() {
+			return "Poly{" + "points=" + points + ", path=" + path + ", origen=" + origen + ", holes=" + holes + '}';
+		}
+	}
+
+	private TextExtrude(String text, Font font, double dir) {
+		this.dir = dir;
+		points = new ArrayList<>();
+		this.text=text;
+		Text textNode = new Text(text);
+		textNode.setFont(font);
+
+		// Convert Text to Path
+		Path subtract = (Path) (Shape.subtract(textNode, new Rectangle(0, 0)));
+		// Convert Path elements into lists of points defining the perimeter
+		// (exterior or interior)
+		subtract.getElements().forEach(this::getPoints);
+
+		// Group exterior polygons with their interior polygons
+		polis.stream().filter(LineSegment::isHole).forEach(hole -> {
+			polis.stream().filter(poly -> !poly.isHole())
+					.filter(poly -> !((Path) Shape.intersect(poly.getPath(), hole.getPath())).getElements().isEmpty())
+					.filter(poly -> poly.getPath().contains(new Point2D(hole.getOrigen().x, hole.getOrigen().y)))
+					.forEach(poly -> poly.addHole(hole));
+		});
+		polis.removeIf(LineSegment::isHole);
+		
```

Comment:
This code came directly from the original library you were using to make the original text extrude. I also noticed that by flattening out this code, the process gets much more readable and understandable.